### DISCUSSION TITLE
Get current RPM of vehicle

### DISCUSSION THREAD

**Discussion by HarishKumar051993** [carla-simulator/carla#4197] [General]
Hi, I'm trying to find a way to get the current revolutions per minute (RPM) of any wheel at a given time via the Python API.
I know I can get the throttle level and the max RPM of the engine via VehiclePhysicsControl, so in theory I could just derive the wheel RPM from there. Due to gearbox properties etc. that value will be pretty inaccurate I guess.

Is there a way to access wheel RPM via the Python API?

**Comment by OlivierDelbeke** [comment]
I tried to do the same some time ago, and my conclusion was that the RPM is not exported by Carla. It's not in the snapshot ; there is also no RPC call to get it ; and therefore also no Python or C++ API.  It is however easily accessible in UE4's blueprints. As a quick and dirty solution, we publish it on MQTT from the blueprints. It would be quite easy to add new RPC + C++ + Python API, but I think that it should be part of the frame snapshot.

**Reply by HarishKumar051993** [reply]
Thanks @OlivierDelbeke for the reply. Can you just let me know how we can access through UE4's blueprint? Im currently using 4.24 version of unreal engine.

**Reply by OlivierDelbeke** [reply]
There is a function called "Get Engine Rotation Speed" that you can call on the "Vehicle Movement".

**Reply by HarishKumar051993** [reply]
@OlivierDelbeke Thanks, but this "Get Engine Rotation Speed" gives me Wheel RPM?

**Reply by OlivierDelbeke** [reply]
@HarishKumar051993 Sorry, I misread (we usually use "RPM" for the engine). For the wheel RPM, you can probably very simply divide the current speed by the wheel radius. You have PythonAPIs for that.

**Reply by HarishKumar051993** [reply]
@OlivierDelbeke Yeah got it. What is the python API we have for calculating?

It will be helpful if i get that.

**Reply by OlivierDelbeke** [reply]
velocity = self.player.get_velocity()
speed = math.sqrt(velocity.x ** 2 + velocity.y ** 2 + velocity.z ** 2)     # in m/s
And carla.WheelPhysicsControl contains the radius of the wheels.

**Reply by HarishKumar051993** [reply]
@OlivierDelbeke Thank you so much. Let me try on this and get back.

**Reply by HarishKumar051993** [reply]
@OlivierDelbeke When i try to get carla.WheelPhysicsControl properties, iam getting same values for al the vehicles(For eg: Radius). How to get the exact vehicle properties of Carla.WheelPhysicsControl ? Any idea on that?

**Reply by James721884** [reply]
@HarishKumar051993 How do you read the wheel radius? By using the following;
`        print (carla.WheelPhysicsControl.radius)`
I get;
`<property object at 0x0000025EFAC8F9F8>`
Is there a way to get it in meters?
